php/php.project/src/org/netbeans/modules/php/project/ComputeTestMethodAnnotations.java

```diff
@@ -142,59 +178,72 @@ public void run() {
         }
     }
 
-    private List<TestMethod> getTestMethods(Document document) {
-        List<TestMethod> testMethods = new ArrayList<>();
-        FileObject fileObject = NbEditorUtilities.getFileObject(document);
-        if (fileObject != null) {
-            PhpProject project = PhpProjectUtils.getPhpProject(fileObject);
-            assert project != null;
-            PhpModule phpModule = project.getPhpModule();
-            for (PhpTestingProvider testingProvider : project.getTestingProviders()) {
+    /**
+     * Get test methods for the first available testing provider.
+     *
+     * @return test methods
+     */
+    private List<TestMethod> getTestMethods() {
+        FileObject fileObject = NbEditorUtilities.getFileObject(handledDocument);
+        if (fileObject == null) {
+            return Collections.emptyList();
+        }
+        PhpProject project = PhpProjectUtils.getPhpProject(fileObject);
+        if (project == null) {
+            return Collections.emptyList();
+        }
+        PhpTestingProvider testingProvider = getFirstTestingProvider(handledDocument);
+        if (testingProvider == null) {
+            return Collections.emptyList();
+        }
+        return getTestMethods(fileObject, testingProvider, project.getPhpModule());
+    }
 
-                if (!document.equals(handledDocument)) {
+    private List<TestMethod> getTestMethods(FileObject fileObject, PhpTestingProvider testingProvider, PhpModule phpModule) {
+        List<TestMethod> testMethods = new ArrayList<>();
+        Collection<PhpClass> phpClasses = getPhpClasses(fileObject);
+        for (PhpClass phpClass : phpClasses) {
+            // check whether the document tab has already been switched to another one
+            if (!handledDocument.equals(getLastFocusedDocument())) {
```

Comment:
Probably, expected condition is not `!document.equals(handledDocument)` but this. Am I wrong??

Reply:
The purpose was to quit fast if the focused document has already changed. Maybe, your code does the same?

Reply:
Returned to the first condition. Thanks!